### DISCUSSION TITLE
CSV Importer: Mixed Content - Resource won't load via HTTPS

### DISCUSSION THREAD

**Discussion by niklas2810** [firefly-iii/firefly-iii#4354] [Firefly III] [answered]
Hey there! 

While trying to set up a new CSV Importer instance (docker container behind Traefik), I noticed that neither CSS  nor JS files would load. I looked at the console window and found the following error:

`Mixed Content: The page at 'https://csv-importer.example.com/token' was loaded over HTTPS, but requested an insecure script 'http://csv-importer.example.com/js/jquery-3.4.1.min.js?version=2.5.0'. This request has been blocked; the content must be served over HTTPS.` (jquery is an example, this error message appears for all CSS/JS files.)

I'm not quite sure why this errors happens, but I guess that the container is not aware of the secure connection. Is there a way for me to tell the CSV importer that all content should be served via HTTPS so that the HTML site is built correctly?

![image](https://user-images.githubusercontent.com/30748216/107125293-47e6c580-68a9-11eb-9690-020ac82b430e.png)

**Comment by niklas2810** [comment, marked as answer]
Oops, I should start reading the docs... Sorry about that!

https://docs.firefly-iii.org/csv/errors/freq_errors/#i-get-page-load-errors-because-the-protocols-dont-match

**Reply by niklas2810** [reply]
I'll try to do the same for Traefik, is there another approach that I could try out?

**Reply by JC5** [reply]
I am not sure for Traefik, but if you know please reply so I can update the docs :D

**Reply by niklas2810** [reply]
@JC5 

I've done some investigation and it seems like Traefik sets all headers automatically (see screenshot). The header which is probably most interesting, `X-Forwarded-Proto`, is set as well. Any clue why the CSV importer doesn't use the headers?

![image](https://user-images.githubusercontent.com/30748216/107127620-98fdb600-68b7-11eb-8425-07d0180c5040.png)

(This is a raw dump of the HTTP request forwarded to the container by replacing the csv-importer with [traefik/whoami](https://github.com/traefik/whoami)).


Edit: FYI - This is the CSV Importer Log.

![image](https://user-images.githubusercontent.com/30748216/107127745-75873b00-68b8-11eb-998a-5bfdfcaf8375.png)

**Comment by Duam** [comment]
I'm having the same issue, but the link to the FAQ points to a 404 page :/ Is there a solution currently? I'm using nginx proxy manager.

**Reply by JC5** [reply]
https://docs.firefly-iii.org/

You can search for "tls", "https" or a variation thereof and you should find your answer.

**Reply by Duam** [reply]
Thanks!

For whomever might stumble upon the same issue, I noticed the old FAQ for this specific issue is still accessible in the history of the docs repository: https://github.com/firefly-iii/docs/blob/6ec5c86a31330c27a54b25fd12f65fc6a212de95/docs/docs/csv/errors/freq_errors.md
